pkg/helm/helm.go

```diff
@@ -0,0 +1,264 @@
+package helm
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"slices"
+	"strconv"
+	"time"
+
+	"github.com/redpanda-data/helm-charts/pkg/kube"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/yaml"
+)
+
+const helmTimestampFormat = `2006-01-02 15:04:05.999999999 -0700 MST`
+
+// Time is a wrapper around [time.Time] to match Helm's JSON time format.
+type Time struct {
+	time.Time
+}
+
+func (t Time) MarshalJSON() ([]byte, error) {
+	return []byte(strconv.Quote(t.Time.Format(helmTimestampFormat))), nil
+}
+
+func (t *Time) UnmarshalJSON(in []byte) error {
+	raw, err := strconv.Unquote(string(in))
+	if err != nil {
+		return err
+	}
+	t.Time, err = time.Parse(helmTimestampFormat, raw)
+	return err
+}
+
+type Repo struct {
+	Name string `json:"name"`
+	URL  string `json:"url"`
+}
+
+type Chart struct {
+	Name        string `json:"name"`
+	Version     string `json:"version"`
+	AppVersion  string `json:"app_version"`
+	Description string `json:"description"`
+}
+
+type Release struct {
+	Name       string `json:"name"`
+	Namespace  string `json:"namespace"`
+	Revision   string `json:"revision"`
+	Updated    Time   `json:"updated"`
+	Status     string `json:"status"`
+	Chart      string `json:"chart"`
+	AppVersion string `json:"app_version"`
+}
+
+// Client is a sandboxed programmatic API for the `helm` CLI.
+//
+// It leverages an isolated HELM_CONFIG_HOME directory to keep operation
+// hermetic but shares a global cache to keep network chatter to a minimum. See
+// `helm env` for more details.
+type Client struct {
+	env []string
+}
+
+type Options struct {
+	ConfigHome string
+	KubeConfig *rest.Config
+}
+
+func (o *Options) asEnv() ([]string, error) {
+	if o.ConfigHome == "" {
+		var err error
+		o.ConfigHome, err = os.MkdirTemp(os.TempDir(), "go-helm-client")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	kubeConfigPath := "/dev/null"
+	if o.KubeConfig != nil {
+		kubeConfigPath = path.Join(o.ConfigHome, "kubeconfig")
+		if err := kube.WriteToFile(kube.RestToConfig(o.KubeConfig), kubeConfigPath); err != nil {
+			return nil, err
+		}
+	}
+
+	return []string{
+		fmt.Sprintf("KUBECONFIG=%s", kubeConfigPath),
+		fmt.Sprintf("HELM_CONFIG_HOME=%s", path.Join(o.ConfigHome, "helm-config")),
+	}, nil
+}
+
+// New creates a new helm client.
+func New(opts Options) (*Client, error) {
+	// Clone the host environment.
+	env, err := opts.asEnv()
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		env: append(os.Environ(), env...),
+	}, nil
+}
+
+type InstallOptions struct {
+	CreateNamespace bool
+	Name            string
+	Namespace       string
+	Values          map[string]any
+	Version         string
+	NoWait          bool
+	NoWaitForJobs   bool
+}
+
+func (o *InstallOptions) asFlags() ([]string, error) {
+	valuesFile, err := os.CreateTemp(os.TempDir(), "helm-values")
```

Comment:
NIT: The temporary file should be deleted after each test.

Reply:
Going to punt on this in this PR. A future commit will handle placing that within the tempDir from `testing.T`